docs/designs/2967-package-image-vulnerability-scanning.md

```diff
@@ -0,0 +1,635 @@
+the attacker this advantage. This strategy of finding a middle ground in doing
+security publicly and not sharing critical vulnerability publicly was used in
+k/k vulnerability scanning after discussions between SIG release, SIG
+architecture and SIG Security.
+
+Other potential option could be limiting access to testgrid dashboard to
+maintainers similar to dependabot. This will give us a chance to be more verbose
+on sharing which image failed exactly and what CVE(s) was detected.
+
+### Alerting on job failures
+
+The scanning job can fail when a vulnerability is found or when it fails for
+some other reason, e.g. scanner failed for some unknown reason, filtering failed
+or scanner timed out. In both cases, it is useful to get alerted on it. For
+this, we should create a private google group to get alerted on using prow's
+built-in mail based alerting functionality, with members who play the following
+roles:
+
+* Community Edition Maintainers
+* Release Engineering team
+* Security Liaisons
+
+For the sake of redundancy and to limit exposure to sensitive data like fixable
+vulnerabilties, it is recommended that only 2 or 3 people (across different
+timezones) in each role should be added. Ideally having an alias for each of
+these roles, will help in managing / rotating members from this google group.
+
+### Running the prow job periodically
+
+Prow has built-in support to configure via yaml the exact interval for
+successive job runs. In k/k a similar vulnerability scan is run every 6 hours.
+So we could use the same time duration for running this job too.
+
+### User Stories
+
+#### Story 1
+
+As a community-edition maintainer, I want to be aware of vulnerabilities in
+tanzu package images, so that I can bump the package versions that fix the
+vulnerabilities or reach out to repo maintainer of the package to get it fixed
+
+#### Story 2
+
+As a community-edition maintainer, I want to restrict the information about
+known vulnerabilities in package images, so that an attacker does not have
+access to this information at the same time as me
+
+#### Story 3
+
+As a community edition maintainer, I want information about vulnerabilities
+close to the code and testing environment, so that it can be fixed with the same
+level of ease as bug fixes instead of having to triage another out of band means
+of tracking vulnerabilities
+
+#### Story 4
+
+As a community edition maintainer, I want to be alerted only on vulnerabilities
+that are actionable, so that I can focus on the ones that need action instead of
+getting bombarded with poor quality vulnerability alerts that consume time with
+limited benefits to overall security of the project
+
+### Constraints
+
+* Selection of vulnerability scanner should support all the base images
+* No licensing issues are encountered (i.e. not exceeding the limit on free
+  version)
+* Community edition jobs can be run on prow and test-grid. Note: This is not a
```

Comment:
100% agree that getting as much overlap with upstream as possible is desired here.

I think we can accomplish the desired state without access to prow and testgrid in the short term, but should look to use it in the long term.

Reply:
Sounds good. Happy to do it once prow and testgrid is in place unless that is going to take a while and we want to implement this design ASAP.

Reply:
I'm really not sure about the prow implementation timeline, so my vote would be to implement in GitHub Actions for the time being. I don't have an opinion on writing this as a script in an Actions runner or using the Trivy action that John mentioned.

Reply:
Thanks for the feedback. Talking to @navidshaikh and @rajaskakodkar, who are leading the prow effort, it seems like it would be good to begin with Github Action based workflow for now as prow and testgrid may not be available for several weeks from now. 

I am adding some more info, about how to do this as part of Github Actions. We may not be able to exactly reuse the Github Action for trivy from Aqua Security as is, but with slight modifications can definitely run it as a Github Action to do what we need to do.